include/zephyr/net/phy.h

```diff
@@ -99,6 +99,10 @@ typedef void (*phy_callback_t)(const struct device *dev,
  * public documentation.
  */
 __subsystem struct ethphy_driver_api {
+	/** Get MAC address */
+	int (*get_mac_address)(const struct device *dev, uint8_t port,
+			       uint64_t *mac);
```

Comment:
Maybe pass a `struct net_eth_addr`?

```suggestion
			       struct net_eth_addr *mac);
```

Reply:
I think it's even more problematic - @jukkar suggest to pass the pointer to uint8_t buffer. But then the application will have to glue all bytes, which looks more like a job for driver.